doc/source/exclude_patterns.md

```diff
@@ -0,0 +1,45 @@
+# Exclude Pattern Settings 
+
+## Introduction
+
+The optional argument `exclude_patterns` in the `in_toto_run` API, also used by `--exclude` in the ```in-toto-run``` command line tool, applies [pathspec](http://python-path-specification.readthedocs.io) to compile [gitignore-style patterns](https://git-scm.com/docs/gitignore).
+
+Passed in arguments such as the paths and hashes of 'materials' (files before command execution) and 'products' (files after command execution) are not recorded with other information when generating link metadata files. If a pattern matches a directory, all files and subdirectories within that directory are also excluded.
```

Comment:
```suggestion
Artifacts (materials and products) matched by an exclude pattern are not recorded when generating link metadata. If a pattern matches a directory, all files and subdirectories are also excluded.
```

Reply:
Should we provide definitions of materials/products as previously, or is it assumed that users of the feature already know what materials and products are?

Reply:
I personally think it's safe to say they know what materials and products are if they're using exclude patterns.